tempesta_fw/sock_srv.c

```diff
@@ -413,6 +413,8 @@ tfw_sock_srv_disconnect(TfwConn *conn)
 	 */
 	if (atomic_read(&conn->refcnt) != TFW_CONN_DEATHCNT)
 		ret = ss_close_sync(sk, true);
+	else if (atomic_read(&conn->refcnt) == TFW_CONN_DEATHCNT)
```

Comment:
I don't see the reason in second `atomic_read()` operation here. Why not just  simple `else`?